Decode space with proper character

mossuchida opened the pull request:
Please review the code.  For HTTP2, it seems like it is simple string, I have use string methods to replace space.

rvansa commented on core/src/main/java/io/hyperfoil/core/client/netty/Http2Connection.java:
```diff
@@ -131,6 +131,19 @@ public void request(HttpRequest request,
+          String subSecond = "";
+          if (question != -1) {
+              subFirst = request.path.substring(0 , question);
+              subFirst = subFirst.replaceAll(" ", "%20");
```
I've suggested replacing that part with `AppendableCharSequence`, was there an issue with that? Using 2x substring, string concatenation and regexp search (!) through `replaceAll` seems rather excessive from memory allocation POV.

mossuchida replied:
I will try to use AppendableCharSequence instead today.

rvansa commented:
If you don't want to bother with HTTP2 you can leave it as it is in the PR, but please add a testcase for both HTTP1.1 and HTTP2.

mossuchida commented:
@rvansa I have updated HTTP2, also added DecodeTest, however, when I tried to package, mvn hangs at my test HttpDecodeSpaceTest. Would you have a look if it looks okay?  I wanted to test it without creating mock server, however, I could not figure out how to unit test, I copied HttpVersionsTest instead.

rvansa commented:
Thanks, I'll take a look.

rvansa commented:
Have you tried to run the test in an IDE or check the test logs? `core/target/surefire-reports/io.hyperfoil.core.client.HttpDecodeSpaceTest-output.txt` shows pretty clear message:
```
12:06:44,843 ERROR (vert.x-eventloop-thread-1) [i.v.c.i.ContextImpl] Unhandled exception java.lang.IllegalArgumentException: code: -1 (expected: 0+)
        at io.vertx.core.http.impl.Http2ServerResponseImpl.setStatusCode(Http2ServerResponseImpl.java:168)
        at io.hyperfoil.core.client.HttpDecodeSpaceTest.isPathCorrect(HttpDecodeSpaceTest.java:201)
```
(even the first `/ping` test tries to use status -1 which is illegal).

rvansa commented:
Btw when you fix the test please rebase & squash the commits; I've (hopefully) fixed the Github Actions so that it doesn't fail with PRs due to missing Quay credentials.

rvansa commented on core/src/test/java/io/hyperfoil/core/client/HttpDecodeSpaceTest.java:
```diff
@@ -0,0 +1,206 @@
+              subSecond = url.substring(url.lastIndexOf("?") + 1);
+              if (subFirst.contains("+")) {
+                  status = 601;
+              } else if (subFirst.contains("%20")) {
```
This condition is superfluous (status becomes 200 anyway).

mossuchida replied:
I have removed unnecessary if conditions.  Please review.

rvansa commented on core/src/test/java/io/hyperfoil/core/client/HttpDecodeSpaceTest.java:
```diff
@@ -0,0 +1,206 @@
+              if (status == 200) {
+                  if (subSecond.contains("%20")) {
+                      status = 602;
+                  } else if (subSecond.contains("+")) {
```
Same as above.

mossuchida replied:
I have removed unnecessary if conditions.  Please review.
